src/openassetio-python/tests/package/_openassetio/test_pyi.py

```diff
@@ -0,0 +1,76 @@
+#
+#   Copyright 2023 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Smoke test for .pyi stub files
+"""
+# pylint: disable=missing-function-docstring,invalid-name
+# pylint: disable=redefined-outer-name
+import ast
+import os
+import pathlib
+
+import pytest
+
+import openassetio
+
+
+expected_pyi_files = (
+    "__init__.pyi",
+    "access.pyi",
+    "constants.pyi",
+    "errors.pyi",
+    "hostApi.pyi",
+    "log.pyi",
+    "managerApi.pyi",
+    "pluginSystem.pyi",
+    "trait.pyi",
+    "utils.pyi",
+)
+
+
+def test_expected_pyi_files_generated(pyi_dir: pathlib.Path):
+    actual_pyi_files = set(p.name for p in pyi_dir.glob("*") if p.name != "_testutils")
+    assert actual_pyi_files == set(expected_pyi_files)
```

Comment:
[Question] Maybe a stupid question, but does this interact at all with turning off stubs in the sanitizer build, or really any build with stubgen turned off? Are these tests not run in that case?

Reply:
The magic for that is at the bottom of the test script.  There is a fixture that will skip all tests in the module if the expected stub directory is not found _and_ there is no env var signalling that stubs should definitely exist (which is set by CMake for `ctest` runs).